mkt/reviewers/views.py

```diff
@@ -1213,3 +1214,50 @@ class ReviewerScoreViewSet(CORSMixin, MarketplaceView, viewsets.ModelViewSet):
     # queryset to prevent instances with other note_key values from ever being
     # returned.
     queryset = ReviewerScore.objects.filter(note_key=mkt.REVIEWED_MANUAL)
+
+
+@reviewer_required
+def moderatelog(request):
+    form = ModerateLogForm(request.GET)
+    modlog = ActivityLog.objects.editor_events()
+    if form.is_valid():
+        if form.cleaned_data['start']:
+            modlog = modlog.filter(created__gte=form.cleaned_data['start'])
+        if form.cleaned_data['end']:
+            modlog = modlog.filter(created__lt=form.cleaned_data['end'])
+        if form.cleaned_data['search']:
+            modlog = modlog.filter(action=form.cleaned_data['search'].id)
+
+    pager = paginate(request, modlog, 50)
+    data = context(request, form=form, pager=pager, tab='moderatelog')
+    return render(request, 'reviewers/moderatelog.html', data)
+
+
+@reviewer_required
+def moderatelog_detail(request, id):
+    log = get_object_or_404(ActivityLog.objects.editor_events(), pk=id)
+    review = None
+    if len(log.arguments) > 1 and isinstance(log.arguments[1], Review):
+        review = log.arguments[1]
+
+    form = ModerateLogDetailForm(request.POST or None)
+    is_admin = acl.action_allowed(request, 'ReviewerAdminTools', 'View')
+    can_undelete = review and review.deleted and (
+                    is_admin or request.user.pk == log.user.pk)
+
+    if (request.method == 'POST' and form.is_valid() and
+        form.cleaned_data['action'] == 'undelete'):
+        if not can_undelete:
+            if not review:
+                raise RuntimeError('Review doesn`t exist.')
```

Comment:
They're catching cases that shouldn't exist - a POST submission attempting to undelete something that isn't deleted or doesn't exist.  If zamboni is rendering these forms its a bug somewhere else so we'd want it in sentry I figured.
I could just take out these lines and award_moderation_points() or review.undelete() would fail instead, at the expense of a slightly longer stacktrace?